### ISSUE TITLE
Check fall-through in _codecs_iso2022.c

### ISSUE THREAD

**Issue by 5531d0d8-2a9c-46ba-8b8b-ef76132a492c** [python/cpython#75458]
BPO | [31275](https://bugs.python.org/issue31275)
--- | :---
Nosy | @vstinner, @skrah, @serhiy-storchaka
PRs | <li>python/cpython#3206</li><li>python/cpython#3518</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-08-25.16:32:52.091>
created_at = <Date 2017-08-25.12:21:15.652>
labels = ['build', '3.7']
title = 'Check fall-through in _codecs_iso2022.c'
updated_at = <Date 2017-09-12.23:15:33.807>
user = 'https://github.com/skrah'
```

bugs.python.org fields:
```python
activity = <Date 2017-09-12.23:15:33.807>
actor = 'vstinner'
assignee = 'none'
closed = True
closed_date = <Date 2017-08-25.16:32:52.091>
closer = 'skrah'
components = ['Build']
creation = <Date 2017-08-25.12:21:15.652>
creator = 'skrah'
dependencies = []
files = []
hgrepos = []
issue_num = 31275
keywords = []
message_count = 5.0
messages = ['300834', '300836', '300844', '300846', '302004']
nosy_count = 3.0
nosy_names = ['vstinner', 'skrah', 'serhiy.storchaka']
pr_nums = ['3206', '3518']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'compile error'
url = 'https://bugs.python.org/issue31275'
versions = ['Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by 5531d0d8-2a9c-46ba-8b8b-ef76132a492c** [comment]
The last fall-through warning is in \_codecs_iso2022.c. IMO the
current code is equivalent to this, but I'm no codecs expert at
all:


```diff
diff --git a/Modules/cjkcodecs/_codecs_iso2022.c b/Modules/cjkcodecs/_codecs_iso2022.c
index 1ce4218f30..abf214880f 100644
--- a/Modules/cjkcodecs/_codecs_iso2022.c
+++ b/Modules/cjkcodecs/_codecs_iso2022.c
@@ -807,15 +807,9 @@ jisx0213_encoder(const Py_UCS4 *data, Py_ssize_t *length, void *config)
     case 2: /* second character of unicode pair */
         coded = find_pairencmap((ucs2_t)data[0], (ucs2_t)data[1],
                                 jisx0213_pair_encmap, JISX0213_ENCPAIRS);
-        if (coded == DBCINV) {
-            *length = 1;
-            coded = find_pairencmap((ucs2_t)data[0], 0,
-                      jisx0213_pair_encmap, JISX0213_ENCPAIRS);
-            if (coded == DBCINV)
-                return MAP_UNMAPPABLE;
-        }
-        else
+        if (coded != DBCINV) {
             return coded;
+        /* fall through */
 
     case -1: /* flush unterminated */
         *length = 1;
```

**Comment by serhiy-storchaka** [comment]
This looks equivalent to me.

**Comment by 5531d0d8-2a9c-46ba-8b8b-ef76132a492c** [comment]
New changeset 138753c1b96b5e06a5c5d409fa4cae5e2fe1108b by Stefan Krah in branch 'master':
bpo-31275: Small refactoring to silence a fall-through warning. (bpo-3206)
https://github.com/python/cpython/commit/138753c1b96b5e06a5c5d409fa4cae5e2fe1108b

**Comment by 5531d0d8-2a9c-46ba-8b8b-ef76132a492c** [comment]
Thanks, Serhiy!

**Comment by vstinner** [comment]
New changeset c0e77364ca29df6cfb311e79892955c92bd8e595 by Victor Stinner in branch '3.6':
[3.6] bpo-30923: Silence fall-through warnings included in -Wextra since gcc-7.0 (bpo-3518)
https://github.com/python/cpython/commit/c0e77364ca29df6cfb311e79892955c92bd8e595
